test/unittest/tests/packing/test_intpack.cpp

```diff
@@ -0,0 +1,416 @@
+        REQUIRE(packed[2] == 0x6e); /* 2nd byte of -4242 as a signed 64-bit number in two bytes */
+        REQUIRE(packed[3] == 0);
+        REQUIRE(packed[4] == 0);
+        REQUIRE(packed[5] == 0);
+        REQUIRE(packed[6] == 0);
+        REQUIRE(packed[7] == 0);
+        unpack_negint_and_check(packed, value);
+    }
+}
+
+TEST_CASE("Integer packing functions: __wt_vpack_int and __wt_vunpack_int", "[intpack]")
+{
+    /*
+     * While the code in each SECTION is small, keeping the code in separate SECTIONS makes it
+     * easier to determine which test has failed should any fail.
+     */
+
+    SECTION("pack and unpack 7")
+    {
+        /*
+         * Expected result is 0x80     | 0x07    = 0x87
+         *                    (marker)  (value)
+         */
+        test_pack_and_unpack_int(7, {0x87, 0, 0, 0, 0, 0, 0, 0});
+    }
+
+    SECTION("pack and unpack 42")
+    {
+        /*
+         * 42 = 0x2a
+         *
+         * Expected result is 0x80     | 0x2a    = 0x0aa
+         *                    (marker)  (value)
+         */
+        test_pack_and_unpack_int(42, {0xaa, 0, 0, 0, 0, 0, 0, 0});
+    }
+
+    SECTION("pack and unpack 256")
+    {
+        /*
+         * 256 = 0x100
+         * 256 - (POS_1BYTE_MAX + 1) = 256 - 0x40 = 0x00c0
+         *
+         * Expected result is 0x0c0   | 0x00                = 0x0c0, and  0x0c0
+         *                    (marker)  (top bits of value)               (bottom 8 bits of value)
+         */
+        test_pack_and_unpack_int(256, {0xc0, 0xc0, 0, 0, 0, 0, 0, 0});
+    }
+
+    SECTION("pack and unpack 257")
+    {
+        /*
+         * 257 = 0x101
+         * 257 - (POS_1BYTE_MAX + 1) = 257 - 0x40 = 0x00c1
+         *
+         * Expected result is 0x0c0   | 0x00              = 0x0c0, and  0x0c1
+         *                    (marker)  (top bits of value)             (bottom 8 bits of value)
+         */
+        test_pack_and_unpack_int(257, {0xc0, 0xc1, 0, 0, 0, 0, 0, 0});
+    }
+
+    SECTION("pack and unpack 0x1234")
+    {
+        /*
+         * 0x1234 - (POS_1BYTE_MAX + 1) = 0x1234 - 0x40 = 0x11f4
+         *
+         * Expected result is 0x0c0    | 0x11              = 0x0d1, and  0x0f4
+         *                    (marker)  (top bits of value)              (bottom 8 bits of value)
+         */
+        test_pack_and_unpack_int(0x1234, {0xd1, 0xf4, 0, 0, 0, 0, 0, 0});
+    }
+
+    SECTION("pack and unpack 0x123456789 - won't fit")
+    {
+        std::vector<uint8_t> packed(8, 0);
+        uint8_t *p = packed.data();
+        int64_t value = 0x123456789;
+        /* value won't fit in two bytes */
+        REQUIRE(__wt_vpack_int(&p, 2, value) == ENOMEM);
+        /* value should fit in 8 bytes */
+        REQUIRE(__wt_vpack_int(&p, packed.size(), value) == 0);
+    }
+
+    SECTION("pack and unpack 0x123456789 - should fit")
+    {
+        /* The value that is stored in this case is (0x123456789 - 0x2040) = 0x123454749 */
```

Comment:
np: This could be merged in a multiline comment

Reply:
Thanks, fixed in 117f0bc.